corehq/messaging/smsbackends/telerivet/models.py

```diff
@@ -75,7 +77,20 @@ def send(self, msg, *args, **kwargs):
             'to_number': msg.phone_number,
             'content': msg.text,
             'message_type': MESSAGE_TYPE_SMS,
+            'status_url': "{base_url}{resource}".format(
+                base_url=get_url_base(),
+                resource=reverse('telerivet_message_status', kwargs={'message_id': msg.couch_id}),
+            ),
+            'status_secret': self.config.webhook_secret
         }
+
+        # This can maybe happen higher up in the stream?
+        phone_number_id = msg.phone_number.replace('+', '')
+        related_case = get_case_by_identifier(msg.domain, phone_number_id)
```

Comment:
I don't think there is a guarantee that the phone number will only return a single case here. Do you think there's a way of adding the case id to the `custom_metadata` on the message when it is queued?

Reply:
@proteusvacuum I realised when I read the spec again I over-engineered the solution. It's not necessary to include the case id - we only need to know the delivery status.